x-pack/filebeat/input/etw/input.go

```diff
@@ -198,7 +192,38 @@ func (e *etwInput) Run(ctx input.Context, publisher stateless.Publisher) error {
 }
 
 // buildEvent builds the winlog object.
-func buildEvent(data map[string]any, h etw.EventHeader, session *etw.Session, cfg config) map[string]any {
+func buildEvent(data map[string]any, h etw.EventHeader, session *etw.Session, cfg config) beat.Event {
+
+	winlog := map[string]any{
+		"activity_guid": h.ActivityId.String(),
+		"channel":       fmt.Sprintf("%d", h.EventDescriptor.Channel),
```

Comment:
Use the `strconv` package for this type of conversion. `fmt.Sprintf` will not be as efficient.